.gitignore

```diff
@@ -131,5 +131,7 @@ dmypy.json
 # Pyre type checker
 .pyre/
 
-# IDE
+# IDE & editors
 .idea/
+# Emacs temp files patterns
+\#*\#
```

Comment:
Newline is missing here (also for the official names see [this](https://stackoverflow.com/questions/151945/how-do-i-control-how-emacs-makes-backup-files), you might want to ignore backup files too).  Moreover, [IDEA seems to target Java only](https://www.jetbrains.com/idea/), so you can remove it now if you don't use it for Python anymore.

A side comment: stop `git add .`.  Always see `status` and `diff` before `commit`.  Use `git commit -a` if possible (it's still risky tho), otherwise manually `add` files needed.  If mistake is make, `git rebase -i` (see `man git-rebase`).

Reply:
IntelliJ IDEA is a Java IDE, but for all other JetBrains IDE they make an `.idea` directory. No idea why they name it so. I'll leave it there in case I'll use it in the future.

Reply:
That's alright I did not know about that.